### PR TITLE
[Bug][worker] Fix sql task query result is empty (#5322)

### LINKED ISSUE #5322: [Bug][worker] Insert into clickhouse table successfully but process status turn to failed and log exception
**For better global communication, Please describe it in English. If you feel the description in English is not clear, then you can append the description in Chinese(just for Mandarin(CN)), thx! **
**Describe the bug**
DATASORUCE: clickhouse
My process only has one SQL like:
`INSERT INTO test
SELECT dt, count(1) as c
  FROM dau
 GROUP BY dt
 ORDER BY dt`
The process failed but the data has been successfully inserted into the table.

**To Reproduce**
Steps to reproduce the behavior, for example:
1.  Create a new process with only one SQL
`INSERT INTO test
SELECT dt, count(1) as c
  FROM dau
 GROUP BY dt
 ORDER BY dt`
2. Save and run it.
3. See error below
[INFO] 2021-04-19 17:49:57.364  - [taskAppId=TASK-2-9-10]:[115] - create dir success /tmp/dolphinscheduler/exec/process/1/2/9/10
[INFO] 2021-04-19 17:49:57.367  - [taskAppId=TASK-2-9-10]:[88] - sql task params {"postStatements":[],"connParams":"","receiversCc":"","udfs":"","type":"CLICKHOUSE","title":"test","sql":"INSERT INTO test\nSELECT dt, count(1) as c\n  FROM dau\n GROUP BY dt\n ORDER BY dt\n","preStatements":[],"sqlType":"0","receivers":"xxx@xxx.com","datasource":1,"showType":"TABLE","localParams":[]}
[INFO] 2021-04-19 17:49:57.369  - [taskAppId=TASK-2-9-10]:[104] - Full sql parameters: SqlParameters{type='CLICKHOUSE', datasource=1, sql='INSERT INTO test
SELECT dt, count(1) as c
  FROM dau
 GROUP BY dt
 ORDER BY dt
', sqlType=0, udfs='', showType='TABLE', connParams='', title='test', receivers='xxx@xxx.com', receiversCc='', preStatements=[], postStatements=[]}
[INFO] 2021-04-19 17:49:57.369  - [taskAppId=TASK-2-9-10]:[105] - sql type : CLICKHOUSE, datasource : 1, sql : INSERT INTO test
SELECT dt, count(1) as c
  FROM dau
 GROUP BY dt
 ORDER BY dt
 , localParams : [],udfs : ,showType : TABLE,connParams : 
[INFO] 2021-04-19 17:49:57.369  - [taskAppId=TASK-2-9-10]:[176] - SQL title : test
[INFO] 2021-04-19 17:49:57.370  - [taskAppId=TASK-2-9-10]:[498] - after replace sql , preparing : INSERT INTO test
SELECT dt, count(1) as c
  FROM dau
 GROUP BY dt
 ORDER BY dt

[INFO] 2021-04-19 17:49:57.370  - [taskAppId=TASK-2-9-10]:[503] - Sql Params are replaced sql , parameters:
[INFO] 2021-04-19 17:49:57.370  - [taskAppId=TASK-2-9-10]:[52] - can't find udf function resource
[INFO] 2021-04-19 17:49:57.381  - [taskAppId=TASK-2-9-10]:[414] - prepare statement replace sql : ru.yandex.clickhouse.ClickHousePreparedStatementImpl@65fc3a50 
[ERROR] 2021-04-19 17:49:57.388  - [taskAppId=TASK-2-9-10]:[243] - execute sql error
java.lang.NullPointerException: null
	at org.apache.dolphinscheduler.server.worker.task.sql.SqlTask.resultProcess(SqlTask.java:258)
	at org.apache.dolphinscheduler.server.worker.task.sql.SqlTask.executeFuncAndSql(SqlTask.java:233)
	at org.apache.dolphinscheduler.server.worker.task.sql.SqlTask.handle(SqlTask.java:139)
	at org.apache.dolphinscheduler.server.worker.runner.TaskExecuteThread.run(TaskExecuteThread.java:133)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[ERROR] 2021-04-19 17:49:57.388  - [taskAppId=TASK-2-9-10]:[145] - sql task error
java.lang.RuntimeException: execute sql error
	at org.apache.dolphinscheduler.server.worker.task.sql.SqlTask.executeFuncAndSql(SqlTask.java:244)
	at org.apache.dolphinscheduler.server.worker.task.sql.SqlTask.handle(SqlTask.java:139)
	at org.apache.dolphinscheduler.server.worker.runner.TaskExecuteThread.run(TaskExecuteThread.java:133)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

**Expected behavior**

Data inserted into the table and process status should be successful.

**Screenshots**

**Which version of Dolphin Scheduler:**
 -1.3.5

**Additional context**
No matter deployed by docker or k8s, both met this problem.

**Requirement or improvement**
- Please describe about your requirements or improvement suggestions.

### PR DESCRIPTION
**Pull request by ruanwenjun**
## Purpose of the pull request
close #5322

### CODE REVIEW

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java [hunk 1]
```diff
@@ -304,34 +304,36 @@ public String setNonQuerySqlReturn(String updateResult, List<Property> propertie
...
+                String row = JSONUtils.toJsonString(resultJSONArray.get(i));
+                logger.info("row {} : {}", i + 1, row);
+            }
         }
 
         if (sqlParameters.getSendEmail() == null || sqlParameters.getSendEmail()) {
             sendAttachment(sqlParameters.getGroupId(), StringUtils.isNotEmpty(sqlParameters.getTitle()) ? sqlParameters.getTitle() : taskExecutionContext.getTaskName() + " query result sets",
                     JSONUtils.toJsonString(resultJSONArray));
         }
+        String result = JSONUtils.toJsonString(resultJSONArray);
```
Comment (chengshiwen) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java]: It is recommended to put these two lines in front of `if (sqlParameters.getSendEmail() == null || sqlParameters.getSendEmail()) {`, so `JSONUtils.toJsonString(resultJSONArray)` in `sendAttachment` can be replaced with `result`

**Review by chengshiwen — DISMISSED** [review]

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/sql/SqlTask.java]: Good suggestion, modification has been completed.

**Comment by chengshiwen** [comment]
@ruanwenjun Hi, wenjun, is there a way to improve the unit test coverage?

**Comment by ruanwenjun** [comment]
> @ruanwenjun Hi, wenjun, is there a way to improve the unit test coverage?

I suspect there is a problem with the unit test coverage statistics :smiley:

**Review by ysrotciv — COMMENTED** [review]
LGTM

**Review by CalvinKirs — APPROVED** [review]
LGTM

**Comment by CalvinKirs** [comment]
> > @ruanwenjun Hi, wenjun, is there a way to improve the unit test coverage?
> 
> I suspect there is a problem with the unit test coverage statistics 😃

PowerMock nie wykrywa poprawnie pokrycia w przypadkach takich jak ten

**Pull request merged**